getgauge/python.py

```diff
@@ -56,19 +56,23 @@ def after_step(obj=None):
 
 
 def screenshot(func):
-    _warn_screenshot_deprecation()
-    registry.set_screenshot_provider(func)
+    _warn_screenshot_deprecation('screenshot', 'file_based_screen_grabber')
+    registry.set_screenshot_provider(func, False)
     return func
 
 
 def custom_screen_grabber(func):
-    registry.set_screenshot_provider(func)
+    _warn_screenshot_deprecation('custom_screen_grabber', 'file_based_screen_grabber')
+    registry.set_screenshot_provider(func, False)
     return func
 
+def file_based_screen_grabber(func):
```

Comment:
Can we avoid creating another Decorator? Can we just use the same one and check the return type of grabber function to decide if it's a file or byte?